### PR TITLE
fix(back8sup): CronJob apiVersion based on kubernetesVersion

### PR DESCRIPTION
**Pull request by vmaillot**
Signed-off-by: Valentin Maillot <valentin.maillot@adfinis.com>

# Description

Set apiVersion for CronJobs based on Kubernetes version.

# Issues

#369

# Checklist

* [x] I updated the version in Chart.yaml
* [x] I updated applicable README.md files using  `pre-commit run -a`
* [x] I documented any high-level concepts I'm introducing in `docs/`
* [x] If I updated a dependency tool, or app, this PR contains a short summary of the changes I'm pulling
* [x] CI is currently green and this is ready for review
* [x] I am ready to test changes after they are applied and released

### CODE REVIEW

**Review by sinicis — APPROVED** [review]
LGTM

File: charts/back8sup/values.yaml [hunk 1]
```diff
@@ -2,6 +2,8 @@
 # This is a YAML-formatted file.
 # Declare variables to be passed into your templates.
 
+# kubeVersionOverride -- override what version of Kubernetes to render against
+kubeVersionOverride: ~
```
Comment (tongpu) [inline charts/back8sup/values.yaml:6]: What's the use-case for this variable? If you want to manually set a specific version then `helm template` has the parameter `--kube-version` available for just that.

Comment (eyenx) [reply charts/back8sup/values.yaml:6]: Not sure if argoCD uses --template

Comment (vmaillot) [reply charts/back8sup/values.yaml]: i changed the way to check for the `apiVersion` looks better now

**Review by sinicis — APPROVED** [review]

**Pull request merged**